File: app/src/main/java/com/imangazalievm/bubbble/presentation/commons/permissions/PermissionsManager.java
package com.imangazalievm.bubbble.presentation.commons.permissions;

public interface PermissionsManager {

    void requestPermission(Permission permission, PermissionRequestListener listener);

    void requestPermissions(Permission[] permissions, PermissionRequestListener listener);

    boolean checkPermissionGranted(Permission permission);

}

File: app-mvp/app/src/main/java/com/bubbble/presentation/global/permissions/PermissionsManager.kt
package com.bubbble.presentation.global.permissions

import com.afollestad.assent.Permission

interface PermissionsManager {

    suspend fun requestPermission(
        permission: Permission
    ): PermissionResult

    suspend fun requestPermissions(
        vararg permissions: Permission
    ): List<PermissionResult>

    suspend fun checkPermission(
        vararg permissions: Permission
    ): List<PermissionResult>

    suspend fun isGranted(
        vararg permissions: Permission
    ): Boolean

}